e3/anod/context.py

```diff
@@ -364,6 +364,25 @@ def add_action(data, connect_with=None):
             if connect_with is not None:
                 self.connect(connect_with, data)
 
+        def add_dep(deps, dep, dep_instance):
+            """Add a new dependency in a dependencies dict.
+
+            :param deps: the Anod instance ``deps`` attribute
+            :type deps: dict
+            :param dep: a Dependency instance
+            :type dep: Dependency
+            :param dep_instance: the Anod instance loaded for that dependency
+            :type dep_instance: Anod
+            """
+            if dep.local_name in deps:
+                raise AnodError(
+                    origin='expand_spec',
+                    message='Error when adding dependency '
+                    '({}, local_name: {}): '
+                    'A dependency with the same local_name '
+                    'has already been loaded'.format(dep.name, dep.local_name))
```

Comment:
Thinking from a user's perspective, what we mean to say is that a spec has two Dependency objects with the same local_name attribute, right? If correct, perhaps we could phrase the error message using that perspective, rather than from the point of view of the mechanics of how/when this was detected?